bc/users/views.py

```diff
@@ -222,3 +230,138 @@ class RateLimitedPasswordResetView(PasswordResetView):
+        new_email = user_cd["email"]
+        changed_email = old_email != new_email
+        if changed_email:
+            # Email was changed.
+            activation_key = user.get_signed_pk()
+
+            # Send an email to the new and old addresses. New for verification;
+            # old for notification of the change.
+            email: EmailType = emails["email_changed_successfully"]
+            send_mail(
+                email["subject"],
+                email["body"] % (user.username, activation_key),
+                email["from_email"],
+                [new_email],
+            )
+            email = emails["notify_old_address"]
+            send_mail(
+                email["subject"],
+                email["body"] % (user.username, old_email, new_email),
+                email["from_email"],
+                [old_email],
+            )
+            msg = message_dict["email_changed_successfully"]
+            messages.add_message(request, msg["level"], msg["message"])
+            logout(request)
+        else:
+            # if the email wasn't changed, simply inform of success.
+            msg = message_dict["settings_changed_successfully"]
+            messages.add_message(request, msg["level"], msg["message"])
+
+        form.save()
+        return HttpResponseRedirect(reverse("profile_settings"))
+
+    return render(
+        request,
+        "profile/settings.html",
+        {"form": form},
+    )
+
+
+@sensitive_post_parameters("old_password", "new_password1", "new_password2")
+@login_required
+def password_change(request: AuthenticatedHttpRequest) -> HttpResponse:
+    if request.method == "POST":
+        form = PasswordChangeForm(user=request.user, data=request.POST)
+        if form.is_valid():
+            form.save()
+            msg = message_dict["pwd_changed_successfully"]
+            messages.add_message(request, msg["level"], msg["message"])
+            update_session_auth_hash(request, form.user)
+            return HttpResponseRedirect(reverse("password_change"))
+    else:
+        form = PasswordChangeForm(user=request.user)
+    return render(
+        request,
+        "profile/password_form.html",
+        {"form": form},
+    )
+
+
+@login_required
+@ratelimiter_unsafe_10_per_m
+@ratelimiter_unsafe_2000_per_h
+def take_out(request: AuthenticatedHttpRequest) -> HttpResponse:
+    if request.method == "POST":
+        email: EmailType = emails["take_out_requested"]
+        send_mail(
+            email["subject"],
+            email["body"] % (request.user, request.user.email),
+            email["from_email"],
+            email["to"],
+        )
+
+        return HttpResponseRedirect(reverse("take_out_done"))
+
+    return render(
+        request,
+        "profile/take_out.html",
+    )
+
+
+def take_out_done(request: HttpRequest) -> HttpResponse:
+    return render(
+        request,
+        "profile/take_out_done.html",
+    )
+
+
+@sensitive_post_parameters("password")
```

Comment:
Will you do a review of variables that should use this decorator and make sure we're not missing any, please?

Reply:
Sure. I noticed I was using the old `activation_key` variable instead of the new `signed_pk` in the `sensitive_variables` decorator.

Thanks for your comment.